src/test/ui/const-generics/uninferred-consts-during-codegen-2.rs

```diff
@@ -0,0 +1,18 @@
+// run-pass
+
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+use std::fmt;
+
+struct Array<T>(T);
+
+impl<T: fmt::Debug, const N: usize> fmt::Debug for Array<[T; N]> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_list().entries((&self.0 as &[T]).iter()).finish()
+    }
+}
+
+fn main() {
+    println!("{:?}", Array([1, 2, 3]));
```

Comment:
Why use `println!(...)` instead of `format!(...)`  + `assert_eq!(...)`?

Reply:
If actually testing the dynamic semantics isn't important then `println!(...)` shouldn't be used and you can just use an assertion function with a bound instead and have `// compile-pass`.